artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -736,11 +736,11 @@ public RoutingStatus route(final Message message,
          throw new IllegalStateException("Message cannot be routed more than once");
       }
 
-      setPagingStore(message);
+      setPagingStore(context.getAddress(message), message);
 
       AtomicBoolean startedTX = new AtomicBoolean(false);
 
-      final SimpleString address = message.getAddressSimpleString();
+      final SimpleString address = context.getAddress(message);
```

Comment:
Can we not make the call to context.getAddress just the once, and re-use the result throughout the method call?

Reply:
I'm just going for the simple refactoring here.. every call to message.getAddressSimplString() to be calling context.getAddress()

Reply:
Your comment could apply to the previous version as well.. I didn't make any big changes here. nor did want to risk anything else.